### PR TITLE
Unified VSX and S390X code path, remove vpermxor 

### PR DESCRIPTION
**Pull request by easyaspi314**
~~This is not ready to merge, it has only been tested for compile errors at the moment. I don't have an s390x toolchain, just Clang's cross compiler.~~

 - Add s390x support to xxhsum
 - Reuse the VSX path for s390x, they both use similar APIs
 - Replace vec_vsx_ld with memcpy
 - Remove vpermxor hack, it makes things too messy
 - Improve Travis tests, dump CPUID info

### CODE REVIEW

**Comment by easyaspi314** [comment]
Typical GCC and its half-assed intrinsic headers. :frowning:

**Comment by easyaspi314** [comment]
```
./xxhsum 0.7.2 (64-bits s390x big endian), GCC 7.4.0, by Yann Collet 
Sample of 100 KB...        
XXH32                    :     102400 ->    33417 it/s ( 3263.4 MB/s) 
XXH32 unaligned          :     102400 ->    33645 it/s ( 3285.7 MB/s) 
XXH64                    :     102400 ->    58312 it/s ( 5694.5 MB/s) 
XXH64 unaligned          :     102400 ->    58443 it/s ( 5707.3 MB/s) 
XXH3_64b                 :     102400 ->   146270 it/s (14284.2 MB/s) 
XXH3_64b unaligned       :     102400 ->   146926 it/s (14348.2 MB/s) 
XXH3_64b seeded          :     102400 ->   155212 it/s (15157.4 MB/s) 
XXH3_64b seeded unaligne :     102400 ->   150673 it/s (14714.2 MB/s) 
XXH128                   :     102400 ->    95155 it/s ( 9292.5 MB/s) 
XXH128 unaligned         :     102400 ->    93561 it/s ( 9136.8 MB/s) 
XXH128 seeded            :     102400 ->    96384 it/s ( 9412.5 MB/s) 
XXH128 seeded unaligned  :     102400 ->    96077 it/s ( 9382.5 MB/s) 
```

**Comment by easyaspi314** [comment]
So I am investigating the unaligned loads.

`vec_xl` seems to emit a different instruction:
```asm
load_vec_xl:
     vl    %v24, 0(%r2), 3   # aligned 8 bytes (2^3)?
     br     %r14

load_memcpy:
     vl    %v24, 0(%r2)
     br     %r14
```

**Comment by easyaspi314** [comment]
Clang's IR seems to verify this:
```llvm
define <2 x i64> @load_vec_xl(i8* nocapture readonly)
{
    %2 = bitcast i8* %0 to <2 x i64>*
    %3 = load <2 x i64>, <2 x i64>* %2, align 8 ; <<<
    ret <2 x i64> %3
}

define <2 x i64> @load_memcpy(i8* nocapture readonly)
{
    %2 = bitcast i8* %0 to <2 x i64>*
    %3 = load <2 x i64>, <2 x i64>* %2, align 1 ; <<<
    ret <2 x i64> %3
}
```

File: .travis.yml [hunk 1]
```diff
@@ -90,10 +90,16 @@ matrix:
         - CPPFLAGS=-DXXH_VECTOR=4 CFLAGS="-O3 -maltivec -mvsx -mpower8-vector -mcpu=power8" LDFLAGS="-static" make check
 
     - name: IBM s390x compilation and consistency checks
+      dist: bionic
```
Comment (Cyan4973) [inline .travis.yml]: Is there a specific reason to pin `dist` to `bionic` ?

File: .travis.yml [hunk 2]
```diff
@@ -90,10 +90,16 @@ matrix:
...
       arch: s390x
       script:
+        # for info
+        - cat /proc/cpuinfo
```
Comment (Cyan4973) [inline .travis.yml]: good idea ! :+1:

File: .travis.yml (hunk 1 above)
Comment (easyaspi314) [reply .travis.yml]: xenial only has GCC 5.4, which doesn't support s390x vector.

File: .travis.yml (hunk 2 above)
Comment (easyaspi314) [reply .travis.yml]: Some reason it doesn't output it though…

**Comment by easyaspi314** [comment]
I think I am going to remove the permxor optimization.

PPC64EB is a *very* uncommon target, and this minor optimization adds a hundred lines or so.

File: .travis.yml (hunk 2 above)
Comment (easyaspi314) [reply .travis.yml]: If I am not mistaken, adding this to the root of the file will apply to all jobs and fold at the start, right?

```yml
before_install:
   - cat /proc/cpuinfo
```

**Comment by easyaspi314** [comment]
Removing permxor definitely makes things easier to follow.

File: .travis.yml (hunk 2 above)
Comment (Cyan4973) [reply .travis.yml]: Seems so.
Never used it though...

**Comment by easyaspi314** [comment]
Cool, it works!

**Comment by easyaspi314** [comment]
I think this is ready to merge.

**Comment by Cyan4973** [comment]
Great ! Thanks @easyaspi314 for this awesome new target !

**Pull request merged**